### PR TITLE
f32/f64/usize in cosmwasm(and other no f32/f64 envs) reproduction and fixes

### LINKED ISSUE #850: serde_json uses floats that may be disallowed in WASM
## Bug Summary

In https://github.com/anoma/namada we're compiling an IBC transaction code to WASM and found that the usage of `serde_json` includes a number parses that uses floats even when floats are not being used (e.g. from `fn send_transfer_execute` from `crates/ibc/src/applications/transfer/relay/send_transfer.rs`). This is problematic for deterministic WASM execution in which floats are disabled. From this https://github.com/serde-rs/json/issues/567 there's a fork that feature-gates floats at https://github.com/nearprotocol/json that can be used in place

### PR DESCRIPTION
**Pull request by Farhad-Shabani**
Closes: #850
Supersedes: #803

### Remark
- As CI took ~13 mins to run cw-check, set CI to run on releases and the main branch.

______

### PR author checklist:
- [X] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests.
- [X] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

**Comment by Farhad-Shabani** [comment]
Hey @tzemanovic, Could you please take a look at this branch and let me know if it fits your needs? Thanks!

**Comment by tzemanovic** [comment]
> Hey @tzemanovic, Could you please take a look at this branch and let me know if it fits your needs? Thanks!

It does, thank you!

File: crates/ibc/clippy.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+disallowed-types = [
+    # { path = "usize", reason = "variable size" }, # cannot on now, because mocks use it and serde, even if there is no usize in type
+    { path = "f64", reason = "harware dependant" },
```
Comment (romac) [inline crates/ibc/clippy.toml]:
```suggestion
    { path = "f64", reason = "not supported in CosmWasm" },
```

**Review by romac — APPROVED** [review]

File: crates/ibc/clippy.toml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    { path = "f32", reason = "harware dependant" },
```
Comment (romac) [inline crates/ibc/clippy.toml]:
```suggestion
    { path = "f32", reason = "not supported in CosmWasm" },
```

**Pull request merged**